packages/did-provider-ebsi/src/EbsiDidProvider.ts

```diff
@@ -127,4 +155,27 @@ export class EbsiDidProvider extends AbstractIdentifierProvider {
     }
     throw Error('no KMS supplied')
   }
+
+  private assertedKeyType = (args: { key?: IKeyOpts; keyType: KeyType }): KeyType => {
+    if (!args.key?.type) {
+      return args.keyType
+    }
+    return args.key.type
+  }
+
+  private assertedPurposes = (args: { keyType: KeyType; purposes: EbsiPublicKeyPurpose[] }) => {
+    if (args.purposes) {
```

Comment:
Since purposes is required this expression would also be favored over the ones below and as it returns immediately, the rest of the function does nothing. IMO purposes should be optional. Also IMO the construct is a bit odd anyway. For EBSI you cannot say that for instance a Secp256k1 key can be used for an assertionMethod; so why would you allow that anyway. IMO this function should check whether the provided purposes are allowed for the specific key. And seperately a method should check when no purposes have been given (so undefined), then it should set the defaults.

Reply:
It makes sense, broke it in two methods as suggested